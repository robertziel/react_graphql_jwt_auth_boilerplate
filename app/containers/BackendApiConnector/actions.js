import {
  NULLIFY_AUTHENTICATION_CREDENTIALS,
  SET_AUTHENTICATION_TOKEN,
  SET_CURRENT_USER,
} from './constants';

export function nullifyAuthenticationCredentials() {
  return {
    type: NULLIFY_AUTHENTICATION_CREDENTIALS,
  };
}

export function setAuthenticationToken(token) {
  return {
    type: SET_AUTHENTICATION_TOKEN,
    token,
  };
}

export function setCurrentUser(currentUser) {
  return {
    type: SET_CURRENT_USER,
    currentUser,
  };
}
