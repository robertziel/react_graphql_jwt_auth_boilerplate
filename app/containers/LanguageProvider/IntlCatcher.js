import { Component } from 'react';
import PropTypes from 'prop-types';
import { injectIntl, intlShape } from 'react-intl';

let intl;

export function getIntl() {
  return intl;
}

export class IntlCatcher extends Component {
  constructor(props) {
    super(props);

    intl = props.intl;

    this.props = props;
  }

  render() {
    return this.props.children;
  }
}

IntlCatcher.propTypes = {
  children: PropTypes.node.isRequired,
  intl: intlShape.isRequired,
};

export default injectIntl(IntlCatcher);
