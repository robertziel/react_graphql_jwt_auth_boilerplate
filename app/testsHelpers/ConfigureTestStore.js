import { memoryHistory } from 'react-router-dom';
import configureStore from 'configureStore';
import StoreAccessor from 'containers/BackendApiConnector/StoreAccessor';

export default function ConfigureTestStore() {
  const confuguredStore = configureStore({}, memoryHistory);

  StoreAccessor.store = confuguredStore.store;

  this.store = confuguredStore.store;
  this.presistor = confuguredStore.presistor;
}
