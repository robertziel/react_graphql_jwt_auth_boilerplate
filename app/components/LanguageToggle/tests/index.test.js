import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';

import { changeLocale } from 'containers/LanguageProvider/actions';
import LanguageProvider from 'containers/LanguageProvider';
import ConfigureTestStore from 'testsHelpers/ConfigureTestStore';
import { translationMessages } from 'i18n';

import LocaleToggle, { mapDispatchToProps } from '../index';

describe('<LocaleToggle />', () => {
  let store;

  function mountWrapper() {
    return mount(
      <Provider store={store}>
        <LanguageProvider messages={translationMessages}>
          <LocaleToggle />
        </LanguageProvider>
      </Provider>,
    );
  }

  beforeAll(() => {
    store = new ConfigureTestStore().store;
  });

  describe('render', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = mountWrapper();
    });

    it('should match the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('should present the default `en` english language option', () => {
      expect(wrapper.find('option[value="en"]')).not.toBeNull();
    });
  });

  describe('mapDispatchToProps', () => {
    describe('onLanguageToggle', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onLanguageToggle).toBeDefined();
      });

      it('should dispatch changeLocale when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const locale = 'de';
        const evt = { target: { value: locale } };
        result.onLanguageToggle(evt);
        expect(dispatch).toHaveBeenCalledWith(changeLocale(locale));
      });
    });
  });
});
