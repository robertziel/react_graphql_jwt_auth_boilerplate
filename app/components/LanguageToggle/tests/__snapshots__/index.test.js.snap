// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocaleToggle /> render should match the snapshot 1`] = `"<div class=\\"Wrapper-sc-1k1j1l-0 bzvZFs\\"><div class=\\"MuiInputBase-root Toggle-swzcly-0 hwtGRd\\"><div class=\\"MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input\\" tabindex=\\"0\\" role=\\"button\\" aria-haspopup=\\"listbox\\">en</div><input aria-hidden=\\"true\\" tabindex=\\"-1\\" class=\\"MuiSelect-nativeInput\\" value=\\"en\\"><svg class=\\"MuiSvgIcon-root MuiSelect-icon\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M7 10l5 5 5-5z\\"></path></svg></div></div>"`;
